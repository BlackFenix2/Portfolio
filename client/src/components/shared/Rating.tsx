import {
  faFrown,
  faStar as nostar
} from '@fortawesome/fontawesome-free-regular';
import { faStar as star } from '@fortawesome/fontawesome-free-solid';
import Icon from '@fortawesome/react-fontawesome';
import * as React from 'react';

import * as styles from './test.module.css';

interface IProps {
  starCount: number;
}

// tslint:disable-next-line:no-console
console.log(styles);

class Rating extends React.PureComponent<IProps> {
  public getRating = starCount => {
    let stars = [];
    const max = 5;
    if (starCount > 0) {
      for (let index = 1; index <= max; index++) {
        stars = stars.concat(
          starCount >= index ? <Star key={index} /> : <EmptyStar key={index} />
        );
      }
    } else {
      stars = stars.concat(<NoStars key={max} />);
    }
    return stars;
  };

  public render() {
    return <span>{this.getRating(this.props.starCount)}</span>;
  }
}

const Star = () => <Icon icon={star} style={{ color: 'gold' }} />;
const EmptyStar = () => <Icon icon={nostar} />;
const NoStars = () => (
  <span>
    <span className={styles.item}>No Stars!</span>
    <Icon icon={faFrown} />
  </span>
);

export default Rating;
