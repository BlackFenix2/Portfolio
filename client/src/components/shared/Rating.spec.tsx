import * as React from 'react';
import { create } from 'react-test-renderer';
import Rating from './Rating';

describe('Rating', () => {
  for (let index = 0; index <= 6; index++) {
    test(`test rating of ${index}`, () => {
      const component = create(<Rating starCount={index} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  }
});
