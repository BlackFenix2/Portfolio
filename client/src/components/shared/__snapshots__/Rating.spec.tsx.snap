// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating test rating of 0 1`] = `
<span>
  <span>
    <span>
      No Stars!
    </span>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-frown fa-w-16 "
      data-icon="frown"
      data-prefix="far"
      role="img"
      style={Object {}}
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm-80-216c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160-64c-17.7 0-32 14.3-32 32s14.3 32 32 32 32-14.3 32-32-14.3-32-32-32zm-80 128c-40.2 0-78 17.7-103.8 48.6-8.5 10.2-7.1 25.3 3.1 33.8 10.2 8.5 25.3 7.1 33.8-3.1 16.6-19.9 41-31.4 66.9-31.4s50.3 11.4 66.9 31.4c4.8 5.7 11.6 8.6 18.5 8.6 5.4 0 10.9-1.8 15.4-5.6 10.2-8.5 11.5-23.6 3.1-33.8C326 321.7 288.2 304 248 304z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
</span>
`;

exports[`Rating test rating of 1 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`Rating test rating of 2 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`Rating test rating of 3 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`Rating test rating of 4 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="far"
    role="img"
    style={Object {}}
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`Rating test rating of 5 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`Rating test rating of 6 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-star fa-w-18 "
    data-icon="star"
    data-prefix="fas"
    role="img"
    style={
      Object {
        "color": "gold",
      }
    }
    viewBox="0 0 576 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;
